### PR TITLE
Feature/airport

### PR DESCRIPTION
**Pull request by thomas99baxter**

### CODE REVIEW

File: lib/airport.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  def attempt_landing(plane)
+    raise "Too stormy to land" if stormy?
+    raise "Airport full!" if airport_full?
+    plane.flying ? land(plane) : raise("Plane is already landed")
+  end
+
+  def attempt_takeoff(plane)
+    raise "Too stormy to takeoff" if stormy?
+    raise "Airport empty!" if airport_empty?
+    is_landed?(plane) ? takeoff(plane) : raise("Your plane isn't in the airport!")
```
Comment (troysnowden) [inline lib/airport.rb:25]: I like this use of the ternary, very niceee

File: lib/airport.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  end
+
+  private
+
+  def airport_full?
+    @landed_planes.length >= @capacity
```
Comment (troysnowden) [inline lib/airport.rb:31]: can the length of landed planes ever be greater than the capacity? Like it won't push to the array if it is equal to the capacity so shouldn't ever be more than that, assuming you never pre populate it with more than the capacity

File: lib/airport.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  def airport_empty?
+    @landed_planes.length == 0
+  end
+
+  def is_landed?(plane)
+    if @landed_planes.include?(plane) or !plane.flying
+      true
+    else
+      false
+    end
```
Comment (troysnowden) [inline lib/airport.rb:43]: Could use your ternary skillz here.
Also, using or instead of ||??? amateur hour

File: spec/airport_spec.rb [hunk 4]
```diff
@@ -0,0 +1,92 @@
+require "airport"
+require "plane"
+CAPACITY = 20
+
+# this needs to be refactored to use stubs rather than the actual classes
+# but i have ran out of time :()
```
Comment (troysnowden) [inline spec/airport_spec.rb:6]: this is redundant now aye?

File: spec/airport_spec.rb [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  it "should raise an error when trying to land an already landed plane" do
+    @test_airport.attempt_landing(@test_plane)
+    expect(@test_airport.landed_planes).to eq([@test_plane])
+
+    expect { @test_airport.attempt_landing(@test_plane) }.to raise_error "Plane is already landed"
+  end
+
+  it "should raise an error when trying to land a plane at a full airport" do
+    CAPACITY.times do 
+      # TODO: stub this
```
Comment (troysnowden) [inline spec/airport_spec.rb:39]: what you wanting to stub here?

File: lib/airport.rb (hunk 2 above)
Comment (thomas99baxter) [reply lib/airport.rb:31]: yeah, so i did have this as an == but gave it a kick about and thought it'd be better to have >= just incase of future cahoots. should be fine tho

File: lib/airport.rb [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  end
+
+  def takeoff(plane)
+    @landed_planes.delete(plane)
+    plane.flying = true
+  end
+
+  def stormy?
+    @weather.stormy
+  end
```
Comment (troysnowden) [inline lib/airport.rb:58]: Is this worth having, or just calling the method directly on the weather object possibly?

File: lib/airport.rb (hunk 3 above)
Comment (thomas99baxter) [reply lib/airport.rb:43]: happy to change this. think this was introduced in the rubocop commit though so might cause a linting issue

File: spec/airport_spec.rb (hunk 5 above)
Comment (thomas99baxter) [reply spec/airport_spec.rb:39]: i was wanting to use a double rather than the actual plane class

File: lib/weather.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+class Weather
+  attr_reader :stormy
+
+  def initialize
+    @stormy = stormy?
+  end
```
Comment (troysnowden) [inline lib/weather.rb:6]: Do you ever use the initial value of 'stormy'? Or do you just call the stormy method each time? As far as I can tell the latter but I haven't fully checked lol

File: lib/airport.rb (hunk 6 above)
Comment (thomas99baxter) [reply lib/airport.rb:58]: yeah, i agree will stick that in.

File: lib/weather.rb (hunk 7 above)
Comment (thomas99baxter) [reply lib/weather.rb:6]: calling the stormy method on the weather object could change the weather each time its called. so its only being set once and then retrieved by the attr reader

File: lib/airport.rb (hunk 6 above)
Comment (thomas99baxter) [reply lib/airport.rb:58]: calling the stormy method on the weather object could change the weather each time its called. so its only being set once and then retrieved by the attr reader

File: lib/weather.rb (hunk 7 above)
Comment (troysnowden) [reply lib/weather.rb:6]: Righttt, I misunderstood, we've implemented it slightly differently. I've just called the rand method each time whereas you set yours and keep the weather the same. That makes sense all good then

File: lib/airport.rb (hunk 6 above)
Comment (troysnowden) [reply lib/airport.rb:58]: Got ya now

**Pull request closed without merging**